### PR TITLE
Exp/refactored sdk/1

### PR DESCRIPTION
**Pull request by ahmadnazir**

### CODE REVIEW

File: src/connector.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+ * @param resource {string} The resource name
+ *
+ * @return {Promise}
+ */
+function createRequest(method, resource) {
+    if (!isConfigured()) {
+        // @todo: Add link to the documentation if it doesn't change
+        //        http://docs.penneo.com/javascript-sdk#init
+        throw 'Please configure the Penneo SDK before calling any methods.';
+        return false;
```
Comment (jesusoterogomez) [inline src/connector.js]: remove return statement

File: src/connector.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        withCredentials: false, // @todo: why do we need to set this?
+        headers: {
+            'X-WSSE': generateAuthHeader(config.key, config.secret),
+            'Accept-charset': 'utf-8',
+            'Accept': 'application/json'
+        }
+    };
+
+    return new Promise(function(resolve, reject) {
+        request(requestOptions, function(error, data) {
```
Comment (jesusoterogomez) [inline src/connector.js]: request should return (error, response);

File: src/connector.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+function delete_(resource) {
+    return createRequest('DELETE', resource);
+}
+
+/* Exports */
+export default {
+    get: get,
+    post: post,
+    put: put,
+    delete: delete_
```
Comment (jesusoterogomez) [inline src/connector.js]: rename functions to _functionName to avoid reserved words in javascript

**Comment by ahmadnazir** [comment]
@jesusoterogomez Please review

**Pull request merged**